packages/node-core/src/indexer/storeCache/cachePoi.ts

```diff
@@ -71,6 +71,13 @@ export class CachePoiModel implements ICachedModelControl, PoiInterface {
   }
 
   async getPoiBlocksByRange(startHeight: number): Promise<ProofOfIndex[]> {
+    await this.mutex.waitForUnlock();
+    const poiBlocks = await this.plainPoiModel.getPoiBlocksByRange(startHeight);
+    return poiBlocks.length !== 0 ? poiBlocks : [];
```

Comment:
If the length is 0 wont poiBlocks == []
```suggestion
    return poiBlocks;
```